src/main/java/org/sagebionetworks/bridge/sdk/ClientInfo.java

```diff
@@ -92,12 +92,8 @@ public synchronized String toString() {
         } else if (appVersion != null) {
             stanzas.add(Integer.toString(appVersion));
         }
-        if (isNotBlank(device) && isNotBlank(osName)) {
-            stanzas.add(String.format("(%s; %s %s)", device, osName, osVersion));
-        } else if (isNotBlank(device)) {
-            stanzas.add(String.format("(%s)", device));
-        } else if (isNotBlank(osName)){
-            stanzas.add(String.format("(%s; %s)", osName, osVersion));
+        if (isNotBlank(device) && isNotBlank(osName) && osVersion != null) {
+            stanzas.add(String.format("(%s; %s/%s)", device, osName, osVersion));
```

Comment:
Some of these formats now fail tests on the server, so sending nothing unless device, osName and osVersion are all set.